scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,862 @@
+
+    private static final String RESULT_IS = ", result is: ";
+
+    private static final String QUOTE = "'";
+
+    private static final String PREVIOUS_VALUE_WAS = ", previous value was ";
+
+    private static final String EXCEPTION_WHEN_EVALUATING_CLOSURE = "Exception when evaluating closure : ";
+
+    private static final String ON_KEY = " on key ";
+
+    private static final String WITH_VALUE = ", with value ";
+
+    private File statusFile;
+
+    private File statusFileDirectory;
+
+    private static final String DEFAULT_STORE_SESSION_NAME = "SchedulerStore";
+
+    public static final String STATUS_RECORD_NAME = "STORE";
+
+    private HashMap<String, Channel> inMemoryChannels;
+
+    private PrimaryHashMap<String, Channel> persistedChannels;
+
+    private Queue<Request> waitUntilQueue = new ArrayDeque<>();
+
+    private CompilerConfiguration compilerConfiguration;
+
+    private GroovyShell shell;
+
+    protected RecordManager recordManager;
+
+    public AOSynchronization() {
+
+    }
+
+    public AOSynchronization(String statusFileDirectoryPath) {
+        compilerConfiguration = new CompilerConfiguration();
+        compilerConfiguration.setTargetBytecode(CompilerConfiguration.JDK8);
+        shell = new GroovyShell(this.getClass().getClassLoader(), compilerConfiguration);
+        this.statusFileDirectory = new File(statusFileDirectoryPath);
+        if (!statusFileDirectory.exists()) {
+            statusFileDirectory.mkdirs();
+        }
+        statusFile = new File(statusFileDirectory, DEFAULT_STORE_SESSION_NAME);
+        inMemoryChannels = new HashMap<>();
+
+    }
+
+    /**
+     * Initialize the Record Manager.
+     *
+     * Load the persisted channel from disk at configured location.
+     *
+     * Delete the persisted database in case a serious error occurred when loading the database.
+     * @param firstAttempt first attempt at loading the persisted channels
+     * @return true if the operation succeeded
+     * @throws IllegalStateException when an error occurred during database loading, even after cleaning it
+     */
+    private boolean init(boolean firstAttempt) {
+        close();
+        try {
+            logger.info("Loading Record Manager from file : " + statusFile);
+            recordManager = RecordManagerFactory.createRecordManager(statusFile.getCanonicalPath());
+            persistedChannels = recordManager.hashMap(STATUS_RECORD_NAME);
+            logger.info("Content of persisted store : " + persistedChannels);
+            // This empty loop triggers InvalidClassException in case of serial version uid problems
+            for (Map.Entry<String, Channel> tempChannels : persistedChannels.entrySet())
+                ;
+            recordManager.commit();
+            if (!firstAttempt) {
+                logger.info("Loading of job database successful after clean.");
+            }
+        } catch (Throwable e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)